src/main/java/com/esri/samples/localserver/local_server_geoprocessing/LocalServerGeoprocessingController.java

```diff
@@ -162,11 +160,18 @@ protected void handleClearResults() {
   }
 
   /**
-   * Stops the animation and disposes of application resources.
+   * Stops and releases all resources used in application.
    */
   void terminate() {
     if (mapView != null) {
       mapView.dispose();
     }
+
+    // stop local server when jvm shuts down
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {

```

Comment:
@mbcoder Based on this, I think it makes sense to just remove the stop methods from the samples. If local server shuts down when the application closes automatically, we shouldn't tell users to add this redundant code. We have the `LocalServerServices` sample to demonstrate stopping it manually with a button. I think that's probably good enough.